jbmc/src/java_bytecode/java_trace_validation.cpp

```diff
@@ -39,7 +39,7 @@ optionalt<symbol_exprt> get_inner_symbol_expr(exprt expr)
 {
   while(expr.has_operands())
   {
-    expr = expr.op0();
+    expr = to_multi_ary_expr(expr).op0();
```

Comment:
This feels like a hack - would it be accurate to say that we're iterating over expressions whose types we don't know getting their first sub anyway? Can this be valid behaviour?

Reply:
Almost always these things are looking for member, index, typecast, address-of and dereference (or some subset thereof) in search of an underlying symbol. Trouble is, we need to reconstruct what's actually possible at each site and verify we're not breaking it by getting more specific.

Reply:
My hope is that doing the opX protection will make code like this less likely in the future.